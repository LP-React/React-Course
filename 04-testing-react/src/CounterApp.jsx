import PropTypes from 'prop-types';
import { useState } from 'react';


export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    const btnClickMin = () => {
        setCounter((c) => c - 1)
    }
    const btnClickReset = () => {
        setCounter(value)
    }
    const btnClickPlus = () => {
        setCounter((c) => c + 1)
    }

    return <>
        <h1>Counter App</h1>
        <h2>{counter}</h2>
        <button onClick={btnClickMin}>- 1</button>
        <button aria-label='btn-reset' onClick={btnClickReset}>Reset</button>
        <button onClick={btnClickPlus}>+ 1</button>

    </>
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}