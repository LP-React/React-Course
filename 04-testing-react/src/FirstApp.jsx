import PropTypes from 'prop-types';

// const newMessage = {
//     num: 123,
//     text: 'Hola'
// };

// const printMessage = (message) => {
//     return JSON.stringify(message)
// }

export const FirstApp = ({ title, subTitle, name }) => {

    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            {/* <code>{ JSON.stringify(newMessage) }</code> */}
            {/* <h2>{printMessage(newMessage)}</h2> */}
            <h2>{subTitle}</h2>
            <h2>{subTitle}</h2>
            <h3>{name}</h3>
        </>
    )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subTitle: PropTypes.string,
}
FirstApp.defaultProps = {
    name: 'LP-React',
    subTitle: 'No hay subtitulo',
}