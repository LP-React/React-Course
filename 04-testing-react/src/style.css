html, body {
    background-color: #21223a;
    color: white;
    font-family: Helvetica, Arial , sans-serif;
    font-size: 1.3rem;
    padding: 0px;
    margin: 0;
    box-sizing: border-box;
}
body {
    width: 100%;
    height: 100dvh;
    display: flex;
    justify-content: center;
    align-items: center;
}
h1,h2 {
    width: 100%;
    text-align: center;
    margin: 15px 0px;
}

button {
    margin: 10px;
}
#root {
    width: 500px;
    flex-flow: wrap;
    display: flex;
    justify-content: center;
    align-items: center;
}
button {
    padding: 10px 30px;
    font-size: 1.3rem;
    border-radius: 10px;
}