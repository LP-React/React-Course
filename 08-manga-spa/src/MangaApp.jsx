import { AppRouter } from "./router/AppRouter"

export const MangaApp = () => {
    return (
        <>
            <AppRouter />
        </>
    )
}
