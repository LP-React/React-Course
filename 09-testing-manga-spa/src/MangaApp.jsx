import { AuthProvider } from "./auth/context/AuthProvider"
import { AppRouter } from "./router/AppRouter"

export const MangaApp = () => {
    return (
        <AuthProvider>
            <AppRouter />
        </AuthProvider>
    )
}
