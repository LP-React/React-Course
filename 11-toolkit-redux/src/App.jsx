import reactLogo from './assets/react.svg'
import './App.css'
import { useDispatch, useSelector } from 'react-redux'
import { decrement, increment, incrementBy } from './store/slices/counter'

function App() {

    const { counter } = useSelector((state) => state.counter)
    const dispatch = useDispatch()

    return (
        <>
            <div>
                <a href="https://react.dev" target="_blank">
                    <img src={reactLogo} className="logo react" alt="React logo" />
                </a>
            </div>
            <h1>Vite + React</h1>
            <p>{counter}</p>
            <div className="card">
                <button onClick={() => dispatch(increment())}>
                    Increment
                </button>
                <button onClick={() => dispatch(decrement())}>
                    Decrement
                </button>
                <button onClick={() => dispatch(incrementBy(2))}>
                    Increment +2
                </button>
            </div>
        </>
    )
}

export default App
