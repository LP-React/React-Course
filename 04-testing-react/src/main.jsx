import React from 'react';
import ReactDOM from 'react-dom/client';

// import {FirstApp} from './FirstApp';
import { CounterApp } from './CounterApp';
import './style.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <FirstApp/> */}
        <CounterApp value={0} />
    </React.StrictMode>
);