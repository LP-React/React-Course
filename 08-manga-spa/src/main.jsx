import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import { BrowserRouter } from 'react-router-dom'

import './index.css'
import { MangaApp } from './MangaApp'

createRoot(document.getElementById('root')).render(
    <StrictMode>
        <BrowserRouter>
            <MangaApp />
        </BrowserRouter>
    </StrictMode>,
)
