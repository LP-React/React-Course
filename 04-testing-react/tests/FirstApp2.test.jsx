import { render, screen } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

describe('Pruebas en <FirsApp/>', () => {

    const title = 'Hola, soy Laysson';
    const subTitle = 'Soy un subtitulo'

    test('Debe hacer match con el snapshot', () => {
        const { container } = render(
            <FirstApp title={title} />
        )
        expect(container).toMatchSnapshot()
    });
    test('Debe mostrar el mensaje "Hola,soy Laysson"', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByText(title)).toBeTruthy();
    });
    test('Debe mostrar el titulo en un h1', () => {
        render(<FirstApp title={title} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(title)
    });
    test('Debe mostrar el Subtitulo enviado por props', () => {
        render(<FirstApp title={title} subTitle={subTitle} />);
        expect(screen.getAllByText(subTitle).length).toBe(2)
    });
})