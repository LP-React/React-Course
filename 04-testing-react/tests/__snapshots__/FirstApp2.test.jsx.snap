// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <FirsApp/> Debe hacer match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, soy Laysson
  </h1>
  <h2>
    No hay subtitulo
  </h2>
  <h2>
    No hay subtitulo
  </h2>
  <h3>
    LP-React
  </h3>
</div>
`;
